etc/dbus-serialbattery/bms/jkbms_brn.py

```diff
@@ -406,10 +414,34 @@ async def enable_charging(self, c):
         # data is 01 00 00 00 for on  00 00 00 00 for off;
         # the following bytes up to 19 are unclear and changing
         # dynamically -> auth-mechanism?
-        await self.write_register(0x1D, b"\x01\x00\x00\x00", 4, c)
-        await self.write_register(0x1E, b"\x01\x00\x00\x00", 4, c)
-        await self.write_register(0x1F, b"\x01\x00\x00\x00", 4, c)
-        await self.write_register(0x40, b"\x01\x00\x00\x00", 4, c)
+        await self.write_register(0x1D, b"\x01\x00\x00\x00", 4, c, True)
+        await self.write_register(0x1E, b"\x01\x00\x00\x00", 4, c, True)
+        await self.write_register(0x1F, b"\x01\x00\x00\x00", 4, c, True)
+        await self.write_register(0x40, b"\x01\x00\x00\x00", 4, c, True)
+
+    def jk_float_to_hex_little(self, val: float):
+        intval = int(val * 1000)
+        hexval = f'{intval:0>8X}'
+        return bytearray.fromhex(hexval)[::-1]
+
+    async def reset_soc_jk(self, c):
+        # Lowering OVPR / OVP
+        # That will trigger a High Voltage Alert
+        ovpr_trigger = round(self.bms_status["settings"]["cell_ovpr"] - 0.250, 3)
+        ovp_trigger = round(self.bms_status["settings"]["cell_ovp"] - 0.250, 3)
```

Comment:
I would recommend:
```python
opv_trigger = self.get_max_cell_voltage() - 0.05
ovpr_trigger = self.get_max_cell_voltage() - 0.10
```

Reply:
Hmm, good idea, should always fit to the actual situation. I'll take that over and do some testing.